filestream flaky tests

## Flaky Test

* **Test Name:** `TestFileWatchNewDeleteModified`, `TestFileScanner`
* **Link:** https://github.com/elastic/beats/blob/35a76dfe2c3caa19969fde4484792355077b58e2/filebeat/input/filestream/fswatch_test.go
* **Branch:** Reproduced locally, seen in these PRs:
  * https://beats-ci.elastic.co/job/Beats/job/beats/job/PR-21444/3/
  * https://beats-ci.elastic.co/job/Beats/job/beats/job/PR-21482/1/
* **Notes:**
  * Possibly related to #21444
  * Reproduced in linux with `while go test -count=1 ./filebeat/input/filestream/...; do true; done`.

### Stack Trace

```
--- FAIL: TestFileWatchNewDeleteModified (0.00s)
    --- FAIL: TestFileWatchNewDeleteModified/two_modified_files (0.00s)
        fswatch_test.go:210: 
            	Error Trace:	fswatch_test.go:210
            	Error:      	Not equal: 
            	            	expected: input_logfile.FSEvent{NewPath:"path1", OldPath:"path1", Op:0x2, Info:filestream.testFileInfo{path:"path1", size:10, time:time.Time{wall:0xbfd5f554f5d9c463, ext:5008327320, loc:(*time.Location)(0xf80b40)}}}
            	            	actual  : input_logfile.FSEvent{NewPath:"path2", OldPath:"path2", Op:0x2, Info:filestream.testFileInfo{path:"path2", size:10, time:time.Time{wall:0xbfd5f554f5d9c463, ext:5008327320, loc:(*time.Location)(0xf80b40)}}}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,7 +1,7 @@
            	            	 (input_logfile.FSEvent) {
            	            	- NewPath: (string) (len=5) "path1",
            	            	- OldPath: (string) (len=5) "path1",
            	            	+ NewPath: (string) (len=5) "path2",
            	            	+ OldPath: (string) (len=5) "path2",
            	            	  Op: (input_logfile.Operation) 2,
            	            	  Info: (filestream.testFileInfo) {
            	            	-  path: (string) (len=5) "path1",
            	            	+  path: (string) (len=5) "path2",
            	            	   size: (int64) 10,
            	Test:       	TestFileWatchNewDeleteModified/two_modified_files
        fswatch_test.go:210: 
            	Error Trace:	fswatch_test.go:210
            	Error:      	Not equal: 
            	            	expected: input_logfile.FSEvent{NewPath:"path2", OldPath:"path2", Op:0x2, Info:filestream.testFileInfo{path:"path2", size:10, time:time.Time{wall:0xbfd5f554f5d9c463, ext:5008327320, loc:(*time.Location)(0xf80b40)}}}
            	            	actual  : input_logfile.FSEvent{NewPath:"path1", OldPath:"path1", Op:0x2, Info:filestream.testFileInfo{path:"path1", size:10, time:time.Time{wall:0xbfd5f554f5d9c463, ext:5008327320, loc:(*time.Location)(0xf80b40)}}}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,7 +1,7 @@
            	            	 (input_logfile.FSEvent) {
            	            	- NewPath: (string) (len=5) "path2",
            	            	- OldPath: (string) (len=5) "path2",
            	            	+ NewPath: (string) (len=5) "path1",
            	            	+ OldPath: (string) (len=5) "path1",
            	            	  Op: (input_logfile.Operation) 2,
            	            	  Info: (filestream.testFileInfo) {
            	            	-  path: (string) (len=5) "path2",
            	            	+  path: (string) (len=5) "path1",
            	            	   size: (int64) 10,
            	Test:       	TestFileWatchNewDeleteModified/two_modified_files
FAIL
FAIL	github.com/elastic/beats/v7/filebeat/input/filestream	0.018s
```

--- a/filebeat/input/filestream/fswatch_test.go
+++ b/filebeat/input/filestream/fswatch_test.go
@@ -38,7 +38,6 @@ var (
 )
 
 func TestFileScanner(t *testing.T) {
-
 	testCases := map[string]struct {
 		paths         []string
 		excludedFiles []match.Matcher

--- a/filebeat/input/filestream/fswatch_test.go
+++ b/filebeat/input/filestream/fswatch_test.go
@@ -38,7 +38,6 @@ var (
 )
 
 func TestFileScanner(t *testing.T) {
-	t.Skip("Flaky test: https://github.com/elastic/beats/issues/21489")
 
 	testCases := map[string]struct {
 		paths         []string
@@ -89,7 +88,7 @@ func TestFileScanner(t *testing.T) {
 			for p, _ := range files {
 				paths = append(paths, p)
 			}
-			assert.True(t, checkIfSameContents(test.expectedFiles, paths))
+			assert.ElementsMatch(t, paths, test.expectedFiles)
 		})
 	}
 }
@@ -116,25 +115,7 @@ func removeFilesOfScannerTest(t *testing.T) {
 	}
 }
 
-// only handles sets
-func checkIfSameContents(one, other []string) bool {
-	if len(one) != len(other) {
-		return false
-	}
-
-	mustFind := len(one)
-	for _, oneElem := range one {
-		for _, otherElem := range other {
-			if oneElem == otherElem {
-				mustFind--
-			}
-		}
-	}
-	return mustFind == 0
-}
-
 func TestFileWatchNewDeleteModified(t *testing.T) {
-	t.Skip("Flaky test: https://github.com/elastic/beats/issues/21489")
 
 	oldTs := time.Now()
 	newTs := oldTs.Add(5 * time.Second)
@@ -226,10 +207,13 @@ func TestFileWatchNewDeleteModified(t *testing.T) {
 
 			go w.watch(context.Background())
 
-			for _, expectedEvent := range test.expectedEvents {
-				evt := w.Event()
-				assert.Equal(t, expectedEvent, evt)
+			count := len(test.expectedEvents)
+			actual := make([]loginp.FSEvent, count)
+			for i := 0; i < count; i++ {
+				actual[i] = w.Event()
 			}
+
+			assert.ElementsMatch(t, actual, test.expectedEvents)
 		})
 	}
 }

--- a/filebeat/input/filestream/fswatch_test.go
+++ b/filebeat/input/filestream/fswatch_test.go
@@ -115,7 +115,6 @@ func removeFilesOfScannerTest(t *testing.T) {
 }
 
 func TestFileWatchNewDeleteModified(t *testing.T) {
-
 	oldTs := time.Now()
 	newTs := oldTs.Add(5 * time.Second)
 	testCases := map[string]struct {

--- a/filebeat/input/filestream/fswatch_test_non_windows.go
+++ b/filebeat/input/filestream/fswatch_test_non_windows.go
@@ -88,7 +88,7 @@ func TestFileScannerSymlinks(t *testing.T) {
 			for p, _ := range files {
 				paths = append(paths, p)
 			}
-			assert.Equal(t, test.expectedFiles, paths)
+			assert.ElementsMatch(t, test.expectedFiles, paths)
 		})
 	}
 }

